src/modules/dashboard/transformers.ts

```diff
@@ -157,3 +163,57 @@ export const getActionsListData = (
     },
   );
 };
+
+export const getEventsListData = (
+  unformattedEvents?: SubgraphEvents,
+): FormattedEvent[] | undefined =>
+  unformattedEvents?.events?.reduce((processedEvents, event) => {
+    if (event) {
+      const {
+        id,
+        associatedColony: { colonyAddress },
+        transaction: {
+          hash,
+          block: { timestamp },
+        },
+        name,
+        args,
+      } = event;
+      const {
+        agent,
+        domainId,
+        recipient,
+        fundingPotId,
+        metadata,
+        token,
+        paymentId,
+        amount,
+        payoutRemainder,
+        decimals = '18',
+      } = JSON.parse(args);
+      const checksummedColonyAddress = createAddress(colonyAddress);
+      return [
+        ...processedEvents,
+        {
+          id,
+          agent: agent ? createAddress(agent) : null,
+          eventName: formatEventName(name),
+          transactionHash: hash,
+          colonyAddress: checksummedColonyAddress,
+          createdAt: new Date(parseInt(`${timestamp}000`, 10)),
+          displayValues: args,
+          domainId: domainId || null,
+          recipient: recipient
+            ? createAddress(recipient)
+            : checksummedColonyAddress,
+          fundingPot: fundingPotId,
+          metadata,
+          tokenAddress: token ? createAddress(token) : null,
+          paymentId,
+          decimals: parseInt(decimals, 10),
+          amount: amount || payoutRemainder || '0',
+        },
+      ];
+    }
+    return undefined;
```

Comment:
Just to be sure... so if we have array [event1, event2, null, event3, event4], then  getEventsListData will return undefined? is it intended?

Reply:
No. Actually good catch. I think this was an oversight on my part.

This should actually return the current value of the `processedEvents` array.